addons/mail/models/models.py

```diff
@@ -46,6 +46,8 @@ def _mail_track(self, tracked_fields, initial):
                     tracking_sequence = 100
                 tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info, tracking_sequence, self._name)
                 if tracking:
+                    if tracking['field_type'] == 'monetary':
+                        tracking['currency_id'] = getattr(self, col_info.get('currency_field', ''), self.company_id.currency_id).id
```

Comment:
If the currency is also changed, it should take the *previous* value.

Maybe the currency_field should be forced to be tracked

Reply:
Not sure. If an amount changes on an SO, what we track is the new one, in the new currency.

Reply:
@KangOl is right though, this whole thing only makes sense if we track "currency before" and "currency after", so we would need 2 extra fields in the table instead of one.

So maybe instead we should just track `currency_id` itself and never bother to store the currency related to a monetary field change. So no extra useless column, and everyone is happy.

Just fetching the data becomes a bit more complex, in that we need to look for what was the applicable currency in the history at the time of the monetary field change.

Reply:
Still think this is not necessary. What is important is the value after the tracking change.

Tracking currency only does not solve the issue. Indeed this means you tracking is not coherent as you have to search through the table to try to find a currency.

I still think keeping the currency of the final tracked value is sufficient for our requirements.